zellij-server/src/route.rs

```diff
@@ -282,47 +289,75 @@ pub(crate) fn route_thread_main(
-                    if route_action(action, rlocked_sessions, &*os_input, &to_server) {
+                    if let Action::SwitchToMode(input_mode) = action {
+                        for client_id in session_state.read().unwrap().clients.keys() {
+                            os_input.send_to_client(
+                                *client_id,
+                                ServerToClientMsg::SwitchToMode(input_mode),
+                            );
+                        }
+                    }
+                    if route_action(action, rlocked_sessions, &*os_input, &to_server, client_id) {
                         break;
                     }
                 }
             }
             ClientToServerMsg::TerminalResize(new_size) => {
+                session_state
+                    .write()
+                    .unwrap()
+                    .set_client_size(client_id, new_size);
+                let min_size = session_state
+                    .read()
+                    .unwrap()
+                    .min_client_terminal_size()
+                    .unwrap();
                 rlocked_sessions
                     .as_ref()
                     .unwrap()
                     .senders
-                    .send_to_screen(ScreenInstruction::TerminalResize(new_size))
+                    .send_to_screen(ScreenInstruction::TerminalResize(min_size))
                     .unwrap();
             }
-            ClientToServerMsg::NewClient(..) => {
-                if *session_state.read().unwrap() != SessionState::Uninitialized {
-                    os_input.send_to_temp_client(ServerToClientMsg::Exit(ExitReason::Error(
-                        "Cannot add new client".into(),
-                    )));
-                } else {
-                    os_input.add_client_sender();
-                    to_server.send(instruction.into()).unwrap();
-                }
+            ClientToServerMsg::NewClient(
+                client_attributes,
+                cli_args,
+                opts,
+                layout,
+                plugin_config,
+            ) => {
+                let new_client_instruction = ServerInstruction::NewClient(
+                    client_attributes,
+                    cli_args,
+                    opts,
+                    layout,
+                    client_id,
+                    plugin_config,
+                );
+                to_server.send(new_client_instruction).unwrap();
             }
-            ClientToServerMsg::AttachClient(_, force, _) => {
-                if *session_state.read().unwrap() == SessionState::Attached && !force {
-                    os_input.send_to_temp_client(ServerToClientMsg::Exit(ExitReason::CannotAttach));
-                } else {
-                    os_input.add_client_sender();
-                    to_server.send(instruction.into()).unwrap();
-                }
+            ClientToServerMsg::AttachClient(client_attributes, force, opts) => {
```

Comment:
The `force` option seems to be obsolete now (It is not being used here or in `ServerInstruction::AttachClient(..)`. Can we remove that?

Reply:
Good catch - fixed. Thanks!